apps/sms/services/activity/service.js

```diff
@@ -0,0 +1,85 @@
+/* global threads */
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Definition of all available private members.
+   * @type {Object}
+   */
+  const priv = Object.freeze({
+    service: Symbol('ActivityService.service'),
+    activityRequest: Symbol('ActivityService.activityRequest'),
+
+    onPostResult: Symbol('ActivityService.onPostResult'),
+    onPostError: Symbol('ActivityService.onPostError'),
+    onActivityRequest: Symbol('ActivityService.onActivityRequest')
+  });
+
+  var ActivityService = function() {
+    var service = this[priv.service] = threads.service(SERVICE_NAME);
+
+    service.method('postResult', this[priv.onPostResult].bind(this));
+    service.method('postError', this[priv.onPostError].bind(this));
+
+    exports.navigator.mozSetMessageHandler(
```

Comment:
yeah, let's not count on this. I think the approach you have here (handling in the main view) is best for the current time.